Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -96,31 +96,108 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, [])
     }
 
-    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewsetMessageOutOfSync() throws {
-        let randomHighValue: UInt64 = 10_1000
+    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewestMessageIsBeyondEndOfNewestMessageButBeforeEndOfFile() throws {
+        let message: TestableMessage = "This is a test"
+        let requiredByteCount = try self.requiredByteCount(for: [message])
+        let maximumBytes = requiredByteCount + 2
```

Comment:
Could this be `1`? Trying to make sure I follow.

Reply:
The answer is no since we want the offset to be before the end of the file.

Reply:
Open to ways to make this more obvious. I was hoping the `+ 2` and `+ 1` in the same test would be reasonably intuitive... but it does require the reader to read the entire method before understanding the magic numbers as-is